src/clr-angular/utils/drag-and-drop/providers/drag-and-drop-event-bus.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+import {ClrDraggableEvent} from "../interfaces/draggable-event";
+
+@Injectable()
+export class ClrDragAndDropEventBus {
+    private dragStart: Subject<ClrDraggableEvent> = new Subject<ClrDraggableEvent>();
+    private dragMove: Subject<ClrDraggableEvent> = new Subject<ClrDraggableEvent>();
+    private dragEnd: Subject<ClrDraggableEvent> = new Subject<ClrDraggableEvent>();
+    private drop: Subject<ClrDraggableEvent> = new Subject<ClrDraggableEvent>();
+
+    get dragStarted(): Observable<ClrDraggableEvent> {
+        return this.dragStart.asObservable();
+    }
+
+    get dragMoved(): Observable<ClrDraggableEvent> {
+        return this.dragMove.asObservable();
+    }
+
+    get dragEnded(): Observable<ClrDraggableEvent> {
+        return this.dragEnd.asObservable();
+    }
+
+    get dropped(): Observable<ClrDraggableEvent> {
+        return this.drop.asObservable();
+    }
+
+    notifyAllOnDragStart(dragEvent: ClrDraggableEvent): void {
```

Comment:
Nit: Prefer a name like `startDrag`, since this is calling the observable with this event.

Reply:
I see. Yeah, that makes more sense. Will change accordingly.

Reply:
Actually, startDrag sounds good. But how about dragMove and dragEnd? I think following method names might work better?
`triggerDragStart()`
`triggerDragMove()`
`triggerDragEnd()`

Reply:
Its not really triggering it, its just a stream of those events. `dragStart()` `dragMove()` `dragEnd()` perhaps?

Reply:
I agree that they are not the source of the events so `trigger*` might not be suitable here. The subjects are already named `dragStart`, `dragMove`,  and `dragEnd`. 

Basically, the main purposes of these public methods are to be called when custom `dragStart`, `dragMove`,  and `dragEnd` events are registered on a draggable and notify all observers of the events. So I think the current names actually suggest their purposes well enough?

Reply:
Maybe `broadcast*` as in `broadcastStartDrag`, since that's what the service is here for?

Reply:
Actually, how about this:

- The `ClrDragEvent` has a `type: EventType` property, where 
```typescript
enum EventType {
  DRAG_START,
  DRAG_MOVE,
  DRAG_END,
  DROP
}
```
- The components properly set the event type on the event when creating it
- `ClrDragAndDropEventBus` offers a single `broadcast(event: ClrDragEvent)` method, that checks the event type and fires the correct Observable.

I like this a lot better than 4 broadcast methods.

Reply:
That's a great idea! 👍